src/myrtlespeech/builders/fully_connected.py

```diff
@@ -36,28 +51,70 @@ def build(
         ...     fully_connected_pb2.FullyConnected()
         ... )
         >>> build(cfg, input_features=32, output_features=16)
-        FullyConnected(
-          (fully_connected): Sequential(
-            (0): Linear(in_features=32, out_features=64, bias=True)
-            (1): ReLU()
-            (2): Linear(in_features=64, out_features=64, bias=True)
-            (3): ReLU()
-            (4): Linear(in_features=64, out_features=16, bias=True)
```

Comment:
We have delegated quite a lot of responsibility to the builders instead of our custom torch.nn.Modules here. In my opinion building the multiple layers should take place in the modules themselves meaning the builders change very little (apart from passing the `batch_norm` bool). So for example the new `FullyConnected` model def would look something like:
```
FullyConnected(
          (fully_connected): Sequential(
            (0): BatchNorm
            (1): Linear(in_features=32, out_features=64, bias=True)
            (2): ReLU()
            etc
```
(i.e. more similar to the original implementation). 

This is quite a big change so it may be best to check with @samgd first. In any case, my reasoning is as follows:

With the proposed design, we don't have full control over the `FullyConnected` output as `Sequential` is the top level module (i.e. we have `Sequential( (0): FullyConnected(...`). Requiring all modules to be sequential may make it harder to add features in future (e.g. parallel filters). 
A concrete example of this is in the `RNN` class, where we use the hidden state outputs for rnnt decoding. With the proposed Sequential design, there isn't a class that will house the functions to collate and parse the hidden state across multiple layers (eg see deepseech_internal: https://github.com/MyrtleSoftware/deepspeech_internal/blob/98420a20b405389fbebbff5afd0dd71e523f2ba7/src/deepspeech/networks/utils.py#L326)

Reply:
Sticking to the standard PyTorch RNN API that includes `num_layers` is a good idea - it means the classes (implementation) can be used in a standalone manner without having to deal with builders/protobufs (e.g. creating an RNN layer in a notebook). 

Keeping the implementation contained in one place may lead to less bugs (single responsibility). What currently happens if `dropout = 0.5` is passed to `LSTM`?

Reply:
I have just changed both the RNN and the FullyConnected building phases. Now everything is done inside the RNN and FullyConnected classes. Their forward functions are now a bit longer because it is necessary to check the type of each single layer in a for loop (whether it is a batch_norm, rnn, etc.) in order to run them with the correct input sizes (or to pad the input sequences)